site/assignments/zombies/zombies.rst

```diff
@@ -131,6 +131,74 @@ Part 6 --- How Many Cities are Infected
+world before the simulation step occurs, and an *after* state, which is the state of the world after the simulation
+step. To do this easily, a copy of the before state can be made (see the provided ``copy_world`` function) such that
+your program has two versions of the world for the execution of the step of the simulation. Information for the
+simulation step is retrieved from the *before* state, and all changes to the world will be put into the *after* state.
+
+Consider the below image as an example. In the *before state*, cities 0 and 2 are infected, and cities 1, 3, and 4 are
+not infected. If during the simulation step city 0 were to become cured, and cities 1 and 4 become infected, the changes
+are made to the *after* state while the *before* state is left alone, unchanged. Again, note that *before state* and
+*after state* are two separate world (two separate lists of cities).
+
+.. image:: before_after_states.png
+
+The simulation of a step requires checking if a city will either spread the disease or cure itself. This is where the
+``spread_probability`` and ``cure_probability`` values come in. These parameters will have some value between 0 -- 1
+--- the closer to 0, the less likely we want the event is to occur, and the closer to 1, the more likely. To achieve
+this functionality, we can make use of Python's ``random`` library, namely, the
+`random function <https://docs.python.org/3/library/random.html#random.random>`_ . This function will return some value
+between 0 and 1 from a uniform distribution. Therefore, calling this function will provide some arbitrary value between
+0 -- 1, and by checking if that value is, for example ``< cure_probability``, we can implement the probabilistic
+occurrences of the events we want. For example, if ``cure_probability`` was set to ``0.80`` (80%), and since
+``random`` is equally likely to select all values between 0 -- 1, ``random`` will produce a value less than ``0.80``
+80% of the time.
+
+When it comes to randomly selecting a neighbour to infect, the ``random`` library provides many functions that may work
+effectively. I suggest investigating the
+`choice function <https://docs.python.org/3/library/random.html#random.choice>`_ as you may find it to be the simplest
+to get working for your needs.
+
+Finally, for the purposes of our simulation, we want to make sure city 0 is always infected. There are a few ways one
+could implement this, but perhaps the simplest is to just update city 0 to infected in the after state before the
+function finishes. That way, if city 0 was cured during the simulation step, it is reset to infected, and if city 0
+happened to not get cured, there is no harm in setting it to infected anyways.
+
+
+    ``Make a copy of the world for the after state``
+
+    ``For each city in the before state:``
+
+        ``If the city is infected``
+
+            ``If the city is infecting a neighbouring city based on the probability value``
```

Comment:
is -> should, here? It's a coin flip to determine, "is" sounds wrong to me

Reply:
Fel... Idk

Just don't think about it.